packages/worker/src/api/controllers/global/email.ts

```diff
@@ -24,13 +23,15 @@ export async function sendEmail(
     invite,
     attachments,
   } = ctx.request.body
-  let user: any
+  let user: User | undefined = undefined
   if (userId) {
     const db = tenancy.getGlobalDB()
-    user = await db.get<User>(userId)
+    user = await db.tryGet<User>(userId)
+    if (!user) {
+      ctx.throw(404, "User not found.")
+    }
```

Comment:
This is where the original bug was. I had this `if (!user)` check outside of the `if (userId)` block, meaning whenever a `userId` wasn't specified (e.g. in automations), it would always return a 404.

This was an oversight on my part that we had no tests to catch. I've added tests that catch this.